pkg/provisioner/ironic/hardwaredetails/hardwaredetails.go

```diff
@@ -110,8 +110,15 @@ func getDiskType(diskdata introspection.RootDiskType) metal3v1alpha1.DiskType {
 func getStorageDetails(diskdata []introspection.RootDiskType) []metal3v1alpha1.Storage {
 	storage := make([]metal3v1alpha1.Storage, len(diskdata))
 	for i, disk := range diskdata {
+		device := disk.Name
+		allNames := []string{device}
+		if disk.ByPath != "" {
+			device = disk.ByPath
```

Comment:
This feels wrong to me, you're replacing `Name` with by-path.

Reply:
That was in the other PR that you already lgtmd :)